src/lib/transaction.ts

```diff
@@ -736,3 +456,327 @@ export async function sendTransaction(
+ *
+ * @internal This is an internal helper function used by sendVersionedTransaction
+ */
+async function sendVersionedTransactionWithRetry(
+  connection: Connection,
+  transaction: VersionedTransaction,
+  {
+    maxRetries = DEFAULT_SEND_OPTIONS.maxRetries,
+    initialDelayMs = DEFAULT_SEND_OPTIONS.initialDelayMs,
+    commitment = DEFAULT_SEND_OPTIONS.commitment,
+    skipPreflight = DEFAULT_SEND_OPTIONS.skipPreflight,
+    onStatusUpdate = (status) => console.log("Transaction status:", status),
+  }: SendTransactionOptions = {},
+): Promise<string> {
+  onStatusUpdate?.({ status: "created" });
+  onStatusUpdate?.({ status: "signed" });
+
+  let signature: string | null = null;
+  let status: SignatureStatus | null = null;
+  let retries = 0;
+  // Setting a minimum to decrease spam and for the confirmation to work
+  let delayBetweenRetries = Math.max(initialDelayMs, 500);
+
+  while (retries < maxRetries) {
+    try {
+      const isFirstSend = signature === null;
+
+      // Send transaction if not sent yet
+      signature = await connection.sendRawTransaction(transaction.serialize(), {
+        skipPreflight,
+        preflightCommitment: commitment,
+        maxRetries: 0,
+      });
+
+      if (isFirstSend) {
+        onStatusUpdate?.({ status: "sent", signature: signature ?? "" });
+      }
+
+      // Poll for confirmation
+      let pollTimeout = delayBetweenRetries;
+      const timeBetweenPolls = 500;
+      while (pollTimeout > 0) {
+        await new Promise((resolve) => setTimeout(resolve, timeBetweenPolls));
+        const response = await connection.getSignatureStatus(signature);
+        if (response?.value) {
+          status = response.value;
+          if (
+            status.confirmationStatus === "confirmed" ||
+            status.confirmationStatus === "finalized"
+          ) {
+            onStatusUpdate?.({ status: "confirmed", result: status });
+            return signature ?? "";
+          }
+        }
+        pollTimeout -= timeBetweenPolls;
+      }
+    } catch (error: unknown) {
+      if (error instanceof Error) {
+        console.log(`Attempt ${retries + 1} failed:`, error.message);
+      } else {
+        console.log(`Attempt ${retries + 1} failed:`, error);
+      }
+    }
+
+    retries++;
+
+    if (retries < maxRetries) {
+      onStatusUpdate?.({ status: "retry", signature: signature ?? null });
+      delayBetweenRetries += RETRY_INTERVAL_INCREASE;
+    }
+  }
+
+  throw new Error(`Transaction failed after ${maxRetries} attempts`);
+}
+
+/**
+ * Creates a new address lookup table and extends it with additional addresses
+ *
+ * @param connection - The Solana connection object
+ * @param sender - The keypair of the transaction sender
+ * @param additionalAddresses - Array of additional addresses to include in the lookup table
+ * @returns A tuple containing the lookup table address and the lookup table account
+ */
+export async function CreateLookupTable(
```

Comment:
camel case function name:
```suggestion
export async function createLookupTable(
```